include/alpaka/core/CallbackThread.hpp

```diff
@@ -34,12 +37,18 @@ namespace alpaka::core
                 m_thread.join();
             }
         }
-        auto submit(Task&& newTask) -> std::future<void>
+
+        template<typename NullaryFunction>
+        auto submit(NullaryFunction&& nf) -> std::future<void>
```

Comment:
In my branch I used yesterday to work on the queue issue I did the following to not wrapp the task in a task if the type already fits

only a suggestion
```C++
        auto submit(Task&& newTask) -> std::future<void>
        {
            auto f = newTask.get_future();
            {
                std::unique_lock<std::mutex> lock{m_mutex};
                m_tasks.emplace(std::move(newTask));
                if(!m_thread.joinable())
                    startWorkerThread();
            }
            m_cond.notify_one();
            return f;
        }

        template<typename F>
        auto submit(F&& f) -> std::future<void>
        {
            auto task = Task{f};
            return submit(std::move(task));
        }
```

Reply:
I can add this change if you prefer. I also had it this way in an earlier version of #1870.

Reply:
Added it.